akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ClusterActorSetImpl.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.internal
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.Terminated
+import akka.actor.typed.scaladsl.Behaviors
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.typed.ClusterShardingSettings
+import akka.cluster.sharding.typed.ClusterShardingSettings.StateStoreModeDData
+import akka.cluster.sharding.typed.ShardingEnvelope
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl.EntityParent
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl.KeepAlivePinger
+import akka.cluster.sharding.typed.scaladsl.ClusterActorSet
+import akka.cluster.sharding.typed.scaladsl.ClusterActorSetSettings
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.sharding.typed.scaladsl.EntityContext
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+
+import scala.concurrent.duration.Duration
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object ClusterActorSetImpl {
+  object EntityParent {
```

Comment:
I decided to make an intermediate parent for the actual set actors so that the pings can be swiftly ignored regardless if the set actor is busy, it could also allow us additional commands without messing with the type of the set actor behavior.

Reply:
Not sure the parent adds much value we need. Can't we use the special `StartEntity` message instead?

Reply:
ok, that is some value that can motivate it.
We should be aware of that by having a parent it will not be possible to send user defined messages to the workers, at least it would be difficult with the types (in Typed).

Reply:
@patriknw The idea was anyways to hide addressing them through sharding from the user, if they want they can use receptionist (or some other such way) to reach them.

`StartEntity` could be fine though. For the pausing stuff see comment below.